inc/saisegmentroute.h

```diff
@@ -149,6 +235,297 @@ typedef sai_status_t (*sai_get_segmentroute_sidlist_attribute_fn)(
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_NEXT_HOP_ID,
+
+    /**
+     * @brief Tunnel id for decapsulation
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_TUNNEL
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     * @validonly SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT4 or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT6 or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT46
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_TUNNEL_ID,
+
+    /**
+     * @brief VRF for decapsulation and specific table lookup functions
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_VIRTUAL_ROUTER
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     * @validonly SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_T or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT4 or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT6 or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT46
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_VRF,
+
+    /**
+     * @brief Attach a counter
+     *
+     * When it is empty, then packet hits won't be counted
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_COUNTER
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_COUNTER_ID,
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_END,
+
+    /** Custom range base value */
+    SAI_LOCAL_SID_ENTRY_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_LOCAL_SID_ENTRY_ATTR_CUSTOM_RANGE_END
+
+} sai_local_sid_entry_attr_t;
+
+/**
+ * @brief Local SID Entry
+ */
+typedef struct _sai_local_sid_entry_t
+{
+    /**
+     * @brief Switch ID
+     *
+     * @objects SAI_OBJECT_TYPE_SWITCH
+     */
+    sai_object_id_t switch_id;
+
+    /**
+     * @brief Virtual Router ID
+     *
+     * @objects SAI_OBJECT_TYPE_VIRTUAL_ROUTER
+     */
+    sai_object_id_t vr_id;
+
+    /**
+     * @brief Length of the Locator part of the SID
+     */
+    sai_uint8_t locator_len;
+
+    /**
+     * @brief Length of the Function part of the SID
+     */
+    sai_uint8_t function_len;
+
+    /**
+     * @brief Length of the Args part of the SID
+     */
+    sai_uint8_t args_len;
+
+    /**
+     * @brief IPv6 prefix for Local SID
+     */
+    sai_ip_prefix_t sid;
```

Comment:
Couple of comments:
- sai_ip_prefix_t may be used for an IPv4 or IPv6 prefix. While in SRv6 we only have IPv6. Hence, wouldn't it make sense to define a sai_ipv6_prefix_t ?
- Additionally: given that we already have the locator_len, function_len, args_len I'm not really sure you really need a prefix here...
- The definition of the prefix should be clarified. For example, I have Locator (64b) + Fun (16b) + Arg (16b). What is the prefix? the /80 or the /96 ?  This could be easily clarified in the comment right before it.

Reply:
@pcamaril All very good points. 1) Earlier, I was thinking about defining two separate attributes - sid and sid_mask of type ip6_t instead of sai_ipv6_prefix_t. Defining a sai_ipv6_prefix_t is also a fine alternative.
2) Yes, prefix_len = locator_len + func_len + args_len and can be considered redundant.

We can discuss it further in the next meeting. If community agrees, I'll revert it back to sai_ipv6_t.